### PR TITLE
enforce min python version

### PR DESCRIPTION
**Pull request by lyoshenka**
## PR Checklist
Please check all that apply to this PR using "x":

- [x] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [x] I have checked that this PR does not introduce a breaking change

## PR Type
What kind of change does this PR introduce?
when running `lbrynet` command, it checks the min python version and errors if python is not compatible

Why is this change necessary?
version before python 3.7 do not check that the version is too low when you `pip install`

- [x] Bugfix
- [ ] Feature
- [ ] Breaking changes (bugfix or feature that introduces breaking changes)
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## Fixes

Issue Number: N/A

### CODE REVIEW

File: lbrynet/extras/cli.py [hunk 1]
```diff
@@ -222,6 +223,12 @@ def ensure_directory_exists(path: str):
 
 
 def main(argv=None):
+    if sys.version_info[0] < min_python_version[0] or \
+            (sys.version_info[0] == min_python_version[0] and (sys.version_info[1] < min_python_version[1] or
+             (sys.version_info[1] == min_python_version[1] and sys.version_info[2] < min_python_version[2]))):
+        raise Exception("Must be using Python >= %d.%d.%d" %
+                        (min_python_version[0], min_python_version[1], min_python_version[2]))
+
```
Comment (eukreign) [inline lbrynet/extras/cli.py]: In Python you can compare tuples and it does the right thing:
```
>>> (3, 6, 0) >= (3, 7, 0)
False
>>> (2, 6, 0) >= (3, 7, 0)
False
>>> (3, 7, 0) >= (3, 7, 0)
True
>>> (3, 7, 1) >= (3, 7, 0)
True
```
All of that code can just be: `sys.version_info >= min_python_version`.

File: setup.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
     long_description_content_type="text/markdown",
     keywords="lbry protocol media",
     license='MIT',
-    python_requires='>=3.6',
+    python_requires='>=%d.%d.%d' % (__min_python_version__[0], __min_python_version__[1], __min_python_version__[2]),
```
Comment (eukreign) [inline setup.py]: `f">={'.'.join(__min_python_version__)}"` is cleaner.

File: lbrynet/extras/cli.py [hunk 3]
```diff
@@ -143,7 +144,7 @@ def _granular_action_groups(self):
 
     def error(self, message):
         self.print_help(argparse._sys.stderr)
-        self.exit(2, '\n'+message+'\n')
+        self.exit(2, '\n' + message + '\n')
```
Comment (eukreign) [inline lbrynet/extras/cli.py]: `f"\n{message}\n"`

**Comment by eukreign** [comment]
According to this: https://stackoverflow.com/a/48777286/376675

As long as you have a recent version of pip tools (which works in python2.7) you should get automatic python checks.

Do we really need to add this custom version check code into lbry?

File: lbrynet/extras/cli.py (hunk 1 above)
Comment (lyoshenka) [reply lbrynet/extras/cli.py]: oh awesome!

**Comment by lyoshenka** [comment]
Here's what I'm seeing (in a fresh venv):

```
$ pip --version
pip 19.1.1 from /home/grin/.virtualenvs/lbry/lib/python3.6/site-packages/pip (python 3.6)
$ python --version
Python 3.6.7
$ pip install -e .
Obtaining file:///home/grin/code/lbry/lbry
Collecting torba (from lbrynet==0.38.0rc2)
...
Successfully built torba
Installing collected packages: multidict, idna, yarl, attrs, typing-extensions, idna-ssl, async-timeout, chardet, aiohttp, pycparser, cffi, asn1crypto, coincurve, pbkdf2, six, cryptography, pylru, plyvel, torba, netifaces, aioupnp, appdirs, certifi, colorama, distro, base58, protobuf, msgpack, ecdsa, pyyaml, docopt, hachoir, lbrynet
  Running setup.py develop for lbrynet
Successfully installed aiohttp-3.5.4 aioupnp-0.0.12 appdirs-1.4.3 asn1crypto-0.24.0 async-timeout-3.0.1 attrs-18.2.0 base58-1.0.0 certifi-2019.3.9 cffi-1.12.1 chardet-3.0.4 coincurve-11.0.0 colorama-0.3.7 cryptography-2.5 distro-1.4.0 docopt-0.6.2 ecdsa-0.13 hachoir-3.0a5 idna-2.8 idna-ssl-1.1.0 lbrynet msgpack-0.6.1 multidict-4.5.2 netifaces-0.10.9 pbkdf2-1.3 plyvel-1.0.5 protobuf-3.6.1 pycparser-2.19 pylru-1.1.0 pyyaml-4.2b1 six-1.12.0 torba-0.5.4 typing-extensions-3.7.2 yarl-1.3.0
$ lbrynet start
Traceback (most recent call last):
  File "/home/grin/.virtualenvs/lbry/bin/lbrynet", line 11, in <module>
    load_entry_point('lbrynet', 'console_scripts', 'lbrynet')()
  File "/home/grin/.virtualenvs/lbry/lib/python3.6/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/home/grin/.virtualenvs/lbry/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2843, in load_entry_point
    return ep.load()
  File "/home/grin/.virtualenvs/lbry/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2434, in load
    return self.resolve()
  File "/home/grin/.virtualenvs/lbry/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2440, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/grin/code/lbry/lbry/lbrynet/extras/cli.py", line 16, in <module>
    from lbrynet.extras.daemon.loggly_handler import get_loggly_handler
  File "/home/grin/code/lbry/lbry/lbrynet/extras/daemon/__init__.py", line 1, in <module>
    from lbrynet.extras.daemon import Components  # register Component classes
  File "/home/grin/code/lbry/lbry/lbrynet/extras/daemon/Components.py", line 15, in <module>
    from lbrynet import utils
  File "/home/grin/code/lbry/lbry/lbrynet/utils.py", line 264, in <module>
    @contextlib.asynccontextmanager
AttributeError: module 'contextlib' has no attribute 'asynccontextmanager'
```

File: setup.py (hunk 2 above)
Comment (lyoshenka) [reply setup.py]: python is nuts :-)

**Comment by lyoshenka** [comment]
wait nvm, didn't update setup.py

**Comment by lyoshenka** [comment]
ok here's all we really need

**Pull request merged**